site/pt-br/guide/effective_tf2.md

```diff
@@ -0,0 +1,287 @@
+layers = [tf.keras.layers.Dense(hidden_size, activation=tf.nn.sigmoid) for _ in range(n)]
+perceptron = tf.keras.Sequential(layers)
+
+# layers[3].trainable_variables => retorna [w3, b3]
+# perceptron.trainable_variables => retorna [w0, b0, ...]
+```
+
+Os modelos/camadas do Keras herdam de `tf.train.Checkpointable` e estão integrados a `@tf.function`, o que permite capturar pontos de verificação diretamente ou exportar SavedModels a partir de objetos do Keras. Você não precisa necessariamente usar a API `.fit()` do Keras para aproveitar essas integrações.
+
+Veja abaixo um exemplo de transferência de aprendizagem que demonstra como o Keras facilita a coleta de um subconjunto de variáveis relevantes. Digamos que você esteja treinando um modelo multicabeças com um tronco compartilhado:
+
+```python
+trunk = tf.keras.Sequential([...])
+head1 = tf.keras.Sequential([...])
+head2 = tf.keras.Sequential([...])
+
+path1 = tf.keras.Sequential([trunk, head1])
+path2 = tf.keras.Sequential([trunk, head2])
+
+# Treinar o conjunto de dados principal
+for x, y in main_dataset:
+  with tf.GradientTape() as tape:
+    # training=True somente é necessário se houver camadas com comportamento
+    # diferente durante treinamento versus inferência (exemplo, Dropout).
+    prediction = path1(x, training=True)
+    loss = loss_fn_head1(prediction, y)
+  # Otimiza simultaneamente os pesos de trunk (tronco) e head1 (cabeça 1).
+  gradients = tape.gradient(loss, path1.trainable_variables)
+  optimizer.apply_gradients(zip(gradients, path1.trainable_variables))
+
+# Faz o ajuste fino da segunda cabeça, reusando o tronco
+for x, y in small_dataset:
+  with tf.GradientTape() as tape:
+    # training=True somente é necessário se houver camadas com comportamento
+    # diferente durante treinamento versus inferência (exemplo, Dropout).
+    prediction = path2(x, training=True)
+    loss = loss_fn_head2(prediction, y)
+  # Otimiza somente os pesos de head2 (cabeça 2), e não os pesos de trunk (tronco)
+  gradients = tape.gradient(loss, head2.trainable_variables)
+  optimizer.apply_gradients(zip(gradients, head2.trainable_variables))
+
+# Você pode publicar somente a computação do trunk (tronco) para que outras pessoas possam reusar.
+tf.saved_model.save(trunk, output_path)
+```
+
+### Combine tf.data.Datasets e @tf.function
+
+Ao iterar sobre dados de treinamento que cabem na memória, fique à vontade para usar a interação comum do Python. Caso contrário, `tf.data.Dataset` é a melhor maneira de transmitir dados de treinamento a partir do disco. Conjuntos de dados são [iteráveis (e não iteradores)](https://docs.python.org/3/glossary.html#term-iterable) e funcionam como qualquer outro iterável do Python no modo Eager. Você pode utilizar totalmente os recursos assíncronos de pré-busca/streaming dos conjuntos de dados ao encapsular seu código em `tf.function()`, que substitui uma iteração do Python pelas operações equivalentes de grafo usando o AutoGraph.
```

Comment:
s/interação/iteração/

(ou talvez dizer "comandos de controle de fluxo" ou algo assim)

Reply:
Could be "... fique à vontade para usar os recursos básicos de iteração do Python."